### PR TITLE
feat: use tfplugindocs and update schema for auto doc generation

### PR DESCRIPTION
**Pull request by matt-FFFFFF**
## Problem statement

Manually crafting documentation is no longer feasible as the provider gets more complex. The risk of misalignments between schema and docs is high.

## Summary

Now that the provider uses the plugin framework, we can generate documentation automatically based on the schema.

## URI Change

This change will modify the name of the generated docs files in line with Hashicorp's best practice - there is no way to avoid this. The provider prefix is removed, so the URI will change:

Before: `/providers/Azure/azapi/latest/docs/resources/azapi_update_resource`
After: `/providers/Azure/azapi/latest/docs/resources/update_resource`

This will mean that search indexes will take a while to update. However, we have to make this change sooner or later and I believe that this the benefits of auto doc generation outweigh the issues caused by a brief period of search crawler updates.

To do this we start using tfplugin docs (as recommended by Hashicorp for all these projects).

## Details

- Modify `main.go` to include `go generate` comments
- Update `GNUMakefile` to include `make docs` command and remove legacy website targets
- Move guide content to `templates` to be copied by `tfplugindocs` tool
- Update resource schema to add `MarkdownDescription` fields
- Create templates for resources and data sources
- Implemented `docstrings` package to standardize messaging and simplify markdown formatting for common schema attributes
- Added custom template for `data_plane_resource` to add table.
- Modify test job to include doc gen checks

## Vendoring

Removed vendor dir as it was not being maintained & I believe 1ES do not recommend vendoring deps

### CODE REVIEW

**Comment by matt-FFFFFF** [comment]
@ms-henglu please can I get your thoughts? If you agree then I will look into the issue with the tflint job.

**Comment by ms-henglu** [comment]
Thanks @matt-FFFFFF ! This looks great, and makes it much easier to maintain the docs. I'll check this PR later. Thanks!

**Comment by stemaMSFT** [comment]
Another little minor detail here; @ms-henglu correct me if I'm wrong but I think tons of examples right now set `schema_validation_enabled = false` and also `response_export_values = ["*"]`. Is there a way for us to improve that?

**Comment by ms-henglu** [comment]
> Another little minor detail here; @ms-henglu correct me if I'm wrong but I think tons of examples right now set `schema_validation_enabled = false` and also `response_export_values = ["*"]`. Is there a way for us to improve that?

Thanks @stemaMSFT , yes, I'll open another PR to improve it.

**Review by ms-henglu — CHANGES_REQUESTED** [review]
Thanks for the PR! LGTM, just one minor suggestion, once it's addressed I'll merge this PR.

File: docs/guides/service_principal_client_certificate.md
Comment (matt-FFFFFF) [reply docs/guides/service_principal_client_certificate.md:87]: Hi @ms-henglu I have fixe this now - apologies, I missed it somehow

**Comment by matt-FFFFFF** [comment]
All changes addressed, also updated README to guide on how to generate docs

**Review by ms-henglu — APPROVED** [review]
Thanks! LGTM 💯

**Pull request merged**
